darts/timeseries.py

```diff
@@ -765,7 +765,9 @@ def from_group_dataframe(
             # store static covariate Series and group DataFrame (without static cov columns)
             splits.append(
                 (
-                    pd.DataFrame([static_cov_vals], columns=static_cov_cols),
+                    pd.DataFrame([static_cov_vals], columns=static_cov_cols).astype(
```

Comment:
For example for integer valued categorical columns (of type pd.CategoricalDType). These got converted to numeric integers. I added this because I had a different approach at the beginning. With the current approach, I think we can remove this.